x/stakeibc/keeper/unbonding_records_get_host_zone_unbondings_msgs_test.go

```diff
@@ -1,226 +1,813 @@
+		// 280 - 110 = 170 unbond remaining
+		// valB has #4 priority - unbond up to capacity at 105
+		{Validator: "valB", UnbondAmount: sdkmath.NewInt(105)},
+		// 170 - 105 = 65 unbond remaining
+		// valG has #5 priority - unbond up to capacity at 25
+		{Validator: "valG", UnbondAmount: sdkmath.NewInt(25)},
+		// 65 - 25 = 40 unbond remaining
+		// valD has #6 priority - unbond up to capacity at 30
+		{Validator: "valD", UnbondAmount: sdkmath.NewInt(30)},
+		// 40 - 30 = 10 unbond remaining
+		// valA has #7 priority - unbond up to remaining
+		{Validator: "valA", UnbondAmount: sdkmath.NewInt(10)},
+	}
 
-	//  define the host zone with total delegation and validators with staked amounts
-	validators := []*stakeibc.Validator{
+	tc := s.SetupTestUnbondFromHostZone(totalWeight, totalStake, totalUnbondAmount, validators)
+	s.CheckUnbondingMessages(tc, expectedUnbondings)
+}
+
+func (s *KeeperTestSuite) TestUnbondFromHostZone_NoDelegationAccount() {
+	// Call unbond on a host zone without a delegation account - it should error
+	invalidHostZone := types.HostZone{
+		ChainId:              HostChainId,
+		DelegationIcaAddress: "",
+	}
+	err := s.App.StakeibcKeeper.UnbondFromHostZone(s.Ctx, invalidHostZone)
+	s.Require().ErrorContains(err, "no delegation account found for GAIA: ICA acccount not found on host zone")
+}
+
+func (s *KeeperTestSuite) TestUnbondFromHostZone_ZeroUnbondAmount() {
+	totalWeight := int64(0)
+	totalStake := sdkmath.ZeroInt()
+	totalUnbondAmount := sdkmath.ZeroInt()
+	tc := s.SetupTestUnbondFromHostZone(totalWeight, totalStake, totalUnbondAmount, []*types.Validator{})
+
+	// Call unbond - it should NOT error since the unbond amount was 0 - but it should short circuit
+	err := s.App.StakeibcKeeper.UnbondFromHostZone(s.Ctx, tc.hostZone)
+	s.Require().Nil(err, "unbond should not have thrown an error - it should have simply ignored the host zone")
+
+	// Confirm no ICAs were sent
+	endSequence, found := s.App.IBCKeeper.ChannelKeeper.GetNextSequenceSend(s.Ctx, tc.delegationPortID, tc.delegationChannelID)
+	s.Require().True(found, "sequence number not found after ica")
+	s.Require().Equal(tc.channelStartSequence, endSequence, "sequence number should stay the same since no messages were sent")
+}
+
+func (s *KeeperTestSuite) TestUnbondFromHostZone_ZeroValidatorWeights() {
+	// Setup the test with all zero-weight validators
+	totalWeight := int64(0)
+	totalStake := sdkmath.NewInt(100)
+	totalUnbondAmount := sdkmath.NewInt(10)
+	validators := []*types.Validator{
+		{Address: "valA", Weight: 0, Delegation: sdkmath.NewInt(25)},
+		{Address: "valB", Weight: 0, Delegation: sdkmath.NewInt(50)},
+		{Address: "valC", Weight: 0, Delegation: sdkmath.NewInt(25)},
+	}
+	tc := s.SetupTestUnbondFromHostZone(totalWeight, totalStake, totalUnbondAmount, validators)
+
+	// Call unbond - it should fail
+	err := s.App.StakeibcKeeper.UnbondFromHostZone(s.Ctx, tc.hostZone)
+	s.Require().ErrorContains(err, "No non-zero validators found for host zone")
```

Comment:
It is a weird case, but why would we want to fail?
Imagine we have validators which do have weights, they have a balance but all of it gets unbonded and sent out.
Then these 3 validators get LSM deposits.  In that case we have no weighted validators with any delegation and we have multiple 0 weight validators which do have some... we should still happily continue working in that case?

Reply:
that is a good point. This was an error that was thrown from within GetTargetAmts. I think the idea is that if we have all 0 weight validators, something is clearly very wrong.

If we were throwing the error from UnbondFromHostZone, I'd agree we should remove it, but I think in this case, I'd lean towards not touching GetTargetAmts cause it's called in other places